tests/cli/main.cpp

```diff
@@ -1593,3 +1589,215 @@ BOOST_FIXTURE_TEST_CASE(cli_use_authorized_transfer, cli_fixture) {
+         fc::optional<graphene::chain::bitasset_options> bit_opts;
+         con.wallet_api_ptr->create_asset("nathan", "BOBCOIN", 5, asset_ops, bit_opts, true);
+      }
+      catch(exception& e)
+      {
+         BOOST_FAIL(e.what());
+      }
+      catch(...)
+      {
+         BOOST_FAIL("Unknown exception creating BOBCOIN");
+      }
+
+      // create a new account for Alice
+      {
+         graphene::wallet::brain_key_info bki = con.wallet_api_ptr->suggest_brain_key();
+         BOOST_CHECK(!bki.brain_priv_key.empty());
+         signed_transaction create_acct_tx = con.wallet_api_ptr->create_account_with_brain_key(bki.brain_priv_key, "alice", 
+               "nathan", "nathan", true);
+         con.wallet_api_ptr->save_wallet_file(con.wallet_filename);
+         // attempt to give alice some bitsahres
+         BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to alice");
+         signed_transaction transfer_tx = con.wallet_api_ptr->transfer("nathan", "alice", "10000", "1.3.0", 
+               "Here are some CORE token for your new account", true);
+      }
+
+      // create a new account for Bob
+      {
+         graphene::wallet::brain_key_info bki = con.wallet_api_ptr->suggest_brain_key();
+         BOOST_CHECK(!bki.brain_priv_key.empty());
+         signed_transaction create_acct_tx = con.wallet_api_ptr->create_account_with_brain_key(bki.brain_priv_key, "bob", 
+               "nathan", "nathan", true);
+         // this should cause resync which will import the keys of alice and bob
+         generate_block(app1);
+         // attempt to give bob some bitsahres
+         BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to Bob");
+         signed_transaction transfer_tx = con.wallet_api_ptr->transfer("nathan", "bob", "10000", "1.3.0", 
+               "Here are some CORE token for your new account", true);
+         con.wallet_api_ptr->issue_asset("bob", "5", "BOBCOIN", "Here are your BOBCOINs", true);
+      }
+
+      BOOST_TEST_MESSAGE("Alice has agreed to buy 3 BOBCOIN from Bob for 3 BTS. Alice creates an HTLC");
+      // create an HTLC
+      std::string preimage_string = "My Super Long Secret that is larger than 50 charaters. How do I look?\n";
+      fc::hash160 preimage_md = fc::hash160::hash(preimage_string);
+      std::stringstream ss;
+      for(size_t i = 0; i < preimage_md.data_size(); i++)
+      {
+         char d = preimage_md.data()[i];
+         unsigned char uc = static_cast<unsigned char>(d);
+         ss << std::setfill('0') << std::setw(2) << std::hex << (int)uc;
+      }
+      std::string hash_str = ss.str();
+      BOOST_TEST_MESSAGE("Secret is " + preimage_string + " and hash is " + hash_str);
+      uint32_t timelock = fc::days(1).to_seconds();
+      graphene::chain::signed_transaction result_tx 
+            = con.wallet_api_ptr->htlc_create("alice", "bob", 
+            "3", "1.3.0", "HASH160", hash_str, preimage_string.size(), timelock, "Alice to Bob", true);
```

Comment:
We could change `preimage_string.size()` to `0` to test the HF code.

Reply:
Should tests at the API level demonstrate best practices? Zero preimage size is tested as part of chain_test, and there is no logic at the API level that was changed that I am aware of.

Reply:
Makes sense. Thanks.